server/audit_math/minerva.py

```diff
@@ -1,11 +1,111 @@
+# pylint: disable=invalid-name
 """
-An implementation of Minerva
+Library for performing a Minerva-style ballot polling risk-limiting audit,
+as described by Zagórski et al https://arxiv.org/abs/2008.02315
+
+Note that this library works for one contest at a time, as if each contest being
+targeted is being audited completely independently.
+
+TODO: ensure the no-losers case is handled
+TODO: if necessary pull out risks for individual contests
+
 """
-# TODO: remove the following once we are using the arguments
-# pylint: disable=unused-argument
-from typing import Dict, Tuple, Optional
+from decimal import Decimal
+from collections import defaultdict
+import logging
+from typing import List, Dict, Tuple, Optional
 
+from athena.audit import Audit  # type: ignore
 from .sampler_contest import Contest
+from .shim import minerva_sample_sizes  # type: ignore
+
+# FIXME: make this an environmental variable
+MINERVA_MULTIPLE = 1.5
+
+
+def compute_cumulative_sample(sample_results):
+    """
+    Computes a cumulative sample given a round-by-round sample
+    """
+    cumulative_sample = defaultdict(int)
+    for rd in sample_results:
+        for cand in sample_results[rd]:
+            cumulative_sample[cand] += sample_results[rd][cand]
+    return cumulative_sample
+
+
+def make_arlo_contest(tally, num_winners=1, votes_allowed=1):
```

Comment:
Is this function necessary? Is the constructor for `Contest` not sufficient?

Reply:
There are various circumstances (in doctests, in python notebooks, etc) where I find it handy to have a function that uses defaults and a bit of math to reduce the boilerplate necessary.
I guess another option would be to add something like this to the Contest class.

Reply:
Okay, that's fine then.